firebase-installations/src/main/java/com/google/firebase/installations/CrossProcessLock.java

```diff
@@ -25,67 +26,40 @@
 /** Use file locking to acquire a lock that will also block other processes. */
 class CrossProcessLock {
   private static final String TAG = "CrossProcessLock";
-  private final FileChannel channel;
-  private final FileLock lock;
 
-  private CrossProcessLock(FileChannel channel, FileLock lock) {
-    this.channel = channel;
-    this.lock = lock;
+  interface CrossProcessPersistedInstallation {
```

Comment:
Do we need this interface or can we just use a `Supplier`?

Reply:
Supplier Added in API Level 24

Reply:
I suggest you make it generic at least on the return parameter(and optionally on the exception thrown to support checked exceptions seamlessly), i.e.

```java
interface Producer<T, E extends Throwable> {
  T produce() throws E;
}

static <T, E extends Throwable> T whileLocked(Producer<T, E> producer) {}
```